src/main/java/org/assertj/core/internal/ObjectArrays.java

```diff
@@ -669,6 +673,27 @@ public void assertDoesNotContainNull(AssertionInfo info, Object[] actual) {
     arrays.assertHasOnlyElementsOfTypes(info, failures, actual, types);
   }
 
+  public <E> void assertHasExactlyElementsOfTypes(AssertionInfo info, E[] actual, Class<?>... types) {
+    Objects.instance().assertNotNull(info, actual);
+    List<Object> actualTypeList = extractTypeList(asList(actual));
+    IterableDiff diff = diff(actualTypeList, asList(types), getComparisonStrategy());
+    if (!diff.differencesFound()) {
+      // actual and values have the same types but are they in the same order ?
+      int i = 0;
+      for (Object elementFromActual : actualTypeList) {
```

Comment:
`elementFromActual` is badly named, it should be `elementTypeFromActual` and instead of `Object` it should be a `Class<?>`